### DISCUSSION TITLE
Fetching Data with SWR and related reviews and activities

### DISCUSSION THREAD

**Discussion by mazzelman** [neuefische/questions#819] [Web] [answered]
Hi,

we tried to implement the github login from yesterdays workshop. That is working so far. But we can't figure out how to fetch the activities right. In the app.js we get:

This is what we want and get in app.js:

```
activities: Array(2)
0: {_id: '65ef2b286b7f1c6c6108879a', activityname: 'Beach Relaxation'}
1: {_id: '65ef2b426b7f1c6c6108879b', activityname: 'Snorkeling'}
length: 2
[[Prototype]]: Array(0)
description: "Tropical paradise in French Polynesia."
image: "/images/placeholder.jpg"
```

This is what we get in the detailspage:

```
activities: Array(2)
0: "65ef2a666b7f1c6c61088795"
1: "65ef2aa96b7f1c6c61088796"
length: 2
[[Prototype]]: Array(0)
description: "New York is the most populous city in the United States, located at the southern tip of New York State on one of the world's largest natural harbors."
image: "/images/placeholder.jpg"
```

We need to know, how we can fetch the activitys like this:

activities: Array(2)
0: {_id: '65ef2b286b7f1c6c6108879a', activityname: 'Beach Relaxation'}
1: {_id: '65ef2b426b7f1c6c6108879b', activityname: 'Snorkeling'}

I think we changed something with @onemanwenttomow to get this. But i can't remember what is was. 🥲

**Comment by onemanwenttomow** [comment, marked as answer]
It required a change in how we populate. The user now references favouritePlaces, but each place had references to comments and activities. The way we solved it was as follows. 

``` js
User.findById(id).populate("favouritePlaces")
```

to

``` js
User.findById(id)
  .populate({
    path: 'favouritePlaces', 
    populate: {
      path: 'activities reviews', 
    }
  })
```
